biothings_explorer/call_apis_v2/query.py

```diff
@@ -80,43 +80,99 @@ def get_edge(self, metakg_edge, input_id):
+            if not subject_id:
+                logger.warning("Cannot found subject_id")
+                return []
+
+            for field_name, path in object_fields.items():
+                current_data = item
+                # Navigate through nested structures
+                for part in path:
+                    if isinstance(current_data, dict) and part in current_data:
+                        current_data = current_data[part]
+                    elif isinstance(current_data, list):
+                        current_data = [
+                            obj.get(part) for obj in current_data if part in obj
+                        ]
+                    else:
+                        current_data = None
+                        break
+
+                # Process multiple objects or single object scenarios
+                if isinstance(current_data, list):
+                    for obj in current_data:
+                        formatted_item = self.create_formatted_item(
+                            subject_type,
+                            subject_id,
+                            object_type,
+                            predicate,
+                            field_name,
+                            obj,
+                        )
+                        output.append(formatted_item)
+                elif current_data:
+                    formatted_item = self.create_formatted_item(
+                        subject_type,
+                        subject_id,
+                        object_type,
+                        predicate,
+                        field_name,
+                        current_data,
+                    )
+                    output.append(formatted_item)
+
+        return output
+
+    def determine_subject_id(self, item, possible_keys):
+        """Attempts to determine the subject ID from a list of possible keys"""
+
+        if isinstance(item, str):
+            for key in possible_keys:
+                if key == item:
+                    return key
+        else:
+            for key in possible_keys:
+                if key in item:
+                    return item[key]
+            # If no key directly found, guess based on the first key that looks like an identifier
+            for key in item.keys():
+                if "id" in key or "ID" in key:
+                    return item[key]
+            return  # If nothing suitable is found, return None
+
+    def create_formatted_item(
+        self,
+        subject_type,
+        subject_field,
+        subject_id,
+        object_type,
+        predicate,
+        field_name,
+        object_id,
+    ):
+        """Helper function to create a formatted dictionary item"""
+        return {
+            "subject": {"type": subject_type, subject_field: subject_id},
+            "predicate": predicate,
+            "object": {"type": object_type, field_name: object_id},
+        }
```

Comment:
I left two comments above for `create_formatted_item` function, but here looks like a duplicated function.

Reply:
oh, I forgot to remove these. these methods should be removed and the ones in parser module must be used instead.